src/test/scala/com/databricks/labs/overwatch/pipeline/InitializeTest.scala

```diff
@@ -0,0 +1,224 @@
+        "\"databaseLocation\":\"dbfs:/mnt/overwatch_global/consume/overwatch_etl.db\",\"consumerDatabaseName\":\"overwatch\"" +
+        ",\"consumerDatabaseLocation\":\"dbfs:/mnt/overwatch_global/consume/overwatch.db\"}," +
+        "\"badRecordsPath\":\"/mnt/overwatch/workspace-1029/sparkEventsBadrecords\",\"overwatchScope\":[\"all\"]," +
+        "\"maxDaysToLoad\":60,\"databricksContractPrices\":{\"interactiveDBUCostUSD\":0.56,\"automatedDBUCostUSD\":0.26}," +
+        "\"primordialDateString\":\"2021-01-16\"}"
+
+      val conf = new Config
+      conf.setOrganizationId("dummyorgifid")
+      val init = new Initializer(conf)
+      val validateAndRegisterArgs = PrivateMethod[Initializer]('validateAndRegisterArgs)
+      init invokePrivate validateAndRegisterArgs(incomplete)
+
+      assert(conf.workspaceName == "dummyorgifid")
+      assert(conf.databaseName == "overwatch_etl_test")
+    }
+
+  }
+
+  describe("Tests for dataTargetIsValid function") {
+    import com.databricks.labs.overwatch.utils.DataTarget
+    it("dataTargetIsValid function should throw exception when the current db location is different than the one present already") {
+      val dataTarget = DataTarget(Some("overwatch_etl"),Some("/path/to/database"), Some("/path/prefix"), Some("overwatch"), Some("/path/to/consumer_database"))
+      spark.sql("create database if not exists overwatch_etl")
+
+      val conf = new Config
+      val init = new Initializer(conf)
+      val dataTargetIsValid = PrivateMethod[Boolean]('dataTargetIsValid)
+      assertThrows[BadConfigException](init invokePrivate dataTargetIsValid(dataTarget))
+    }
+
+    it("dataTargetIsValid function should throw exception when the current db is not created from overwatch") {
+      val dataTarget = DataTarget(Some("overwatch_etl"),Some("file:/src/test/resources/overwatch/spark-warehouse/overwatch_etl.db"), Some("/path/prefix"), Some("overwatch"), Some("/path/to/consumer_database"))
+      spark.sql("create database if not exists overwatch_etl location 'file:/src/test/resources/overwatch/spark-warehouse/overwatch_etl.db'")
+      val conf = new Config
+      val init = new Initializer(conf)
+      val dataTargetIsValid = PrivateMethod[Boolean]('dataTargetIsValid)
+      assertThrows[BadConfigException](init invokePrivate dataTargetIsValid(dataTarget))
+    }
+  }
+
+  describe("Tests for validateScope function") {
+    import com.databricks.labs.overwatch.utils.DataTarget
+    it("validateScope function should throw exception when the variable in scope is not one of the valid scope") {
+
+      val conf = new Config
+      val init = new Initializer(conf)
+      val validateScope = PrivateMethod[Seq[OverwatchScope.OverwatchScope]]('validateScope)
+      assertThrows[BadConfigException](init invokePrivate validateScope(Seq("invalidScope")))
+    }
+    it("validateScope function should check if sparkEvents, clusterEvents, and jobs scopes require clusters scope") {
```

Comment:
We need a lot more test cases here:

- I think I see 3 dependency chains here -- we should test them all independently 
- There's at least 1 more where all scopes are valid (positive test case)
- We need case sensitivity tests
- Seq[String] should maintain order, can we validate that? Are there any order dependencies that could break the logic chain?